### PR TITLE
Add docs for `vcpkg edit`

### PR DESCRIPTION
**Pull request by vicroms**

### CODE REVIEW

File: vcpkg/commands/edit.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Synopsis
+
+```Console
+vcpkg edit <ports> [--options]
+```
+
+## Description
+
+Opens a port's corresponding `ports` folder in a new _Visual Studio Code_ window.
```
Comment (ras0219-msft) [inline vcpkg/commands/edit.md:19]: I'm not sure we should introduce the term "`ports` folder" here; I don't believe we use that language elsewhere. I'd instead suggest something like "definition" or avoid it entirely -- "Opens a port for editing in ...".

Comment (ras0219-msft) [inline vcpkg/commands/edit.md:19]: Given that this command doesn't require Visual Studio Code, I don't think its summary should imply that. For example:
"... in an editor window (defaults to Visual Studio Code)".

File: vcpkg/commands/edit.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+The `--buildtrees` option opens the package's `buildtrees` folder instead of its corresponding `ports`
+folder. The `--all` option also opens the port's related `packages`, `installed`, and `buildtrees`
```
Comment (ras0219-msft) [inline vcpkg/commands/edit.md:22]: I don't believe this opens the `installed` folder, but I could be wrong.

File: vcpkg/commands/edit.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+folders in the same window.
+
+The command accepts multiple port names, in which case, all ports are opened in the same window.
```
Comment (ras0219-msft) [inline vcpkg/commands/edit.md:25]:
```suggestion
The command accepts multiple port names. If multiple names are provided, all ports are opened in the same window.
```

File: vcpkg/commands/edit.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+By default, vcpkg will search for _Visual Studio Code_ in well-known installation paths, set the
+`EDITOR` environment variable to add a _Visual Studio Code_ executable not in the usual installation
+paths.
+
+On Windows, vcpkg searches for a _Visual Studio Code_ or _Visual Studio Code - Insiders_ installation in:
```
Comment (ras0219-msft) [inline vcpkg/commands/edit.md:31]: Should we be italicizing product names like Visual Studio and Visual Studio Code? Why not Windows?

File: vcpkg/commands/edit.md (hunk 1 above)
Comment (ras0219-msft) [inline vcpkg/commands/edit.md:19]: We should probably add description covering the circumstances where this command works and doesn't work. For example, I don't know what this command does in the case of custom git or filesystem registries -- off the top of my head, I would assume it actually does work for filesystem registries but gives bogus results for git registries.

File: vcpkg/commands/edit.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ms.date: 7/24/2023
+ms.prod: vcpkg
+---
+# vcpkg edit
+
+## Synopsis
+
+```Console
+vcpkg edit <ports> [--options]
+```
```
Comment (JavierMatosD) [inline vcpkg/commands/edit.md:15]: Would it make sense to list the options in the synopsis for completeness?

i.e., `vcpkg edit <ports> [--all] [--buildtrees]`

File: vcpkg/commands/edit.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+Opens the port's related `buildtrees`, `packages`, and `installed` folders.
+
+### `--buildtrees`
+
+Opens the port's `buildtrees` folder instead of the port's `ports` folder. The `buildtrees` folder
+is created during the installation process and contains the package's extracted source code and log files.
+
+Use in combination with the `vcpkg install --editable <port>` command to get a clean copy of the
+package's source code suitable for creating patch files.
```
Comment (JavierMatosD) [inline vcpkg/commands/edit.md:56]: Would it make sense to explain why `--buildtrees` is useful? I.e., for debugging build issues or for developing and testing patches? Maybe show how `--buildtrees` could be used with `vcpkg install --editable <port>`  or give an example?

[the remark this answers is not in the document]
Comment (vicroms) [reply vcpkg/commands/edit.md:56]: I was under the impression that `--packages` existed, but it doesn't.

Comment (vicroms) [reply vcpkg/commands/edit.md:56]: I think that example belongs in its own article about debugging and patching packages.

File: vcpkg/commands/edit.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+* the _Program Files_ and _Program Files (x86)_ folders,
+* the `%APPDATA%\Local\Programs` folder,
+* the Windows Registry.
+
+On Linux, vcpkg searches for the _Visual Studio Code_ executable:
+
+* `/usr/bin/code`
+* `/usr/share/code/bin/code`
+
+On MacOS, vcpkg searches for a _Visual Studio Code_ or _Visual Studio Code - Insiders_ installation in the `Applications` folder.
```
[the remark this answers is not in the document]
Comment (vicroms) [reply vcpkg/commands/edit.md:42]: Apparently, we don't.

File: vcpkg/TOC.yml [hunk 8]
```diff
@@ -2,62 +2,62 @@
   href: index.yml
 - name: Consuming Libraries
   items:
-    - name: Getting Started with Classic mode
```
[the remark this answers is not in the document]
Comment (vicroms) [reply vcpkg/TOC.yml:5]: Got automatically formatted by the Learn extension, disappears if you do "Hide whitespace" in the files page.

File: vcpkg/commands/edit.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+## Description
+
+Opens a port for editing in a text editor window (defaults to Visual Studio Code).
+
+If multiple port names are provided, all ports are opened in the same window.
+This command does not work with ports from external registries, only ports from the built-in
+registry can be opened.
+
+The `--buildtrees` option opens the package's `buildtrees` folder instead of the port's contents.
+The `--all` option also opens the port's related `packages`, and `buildtrees` folders in the same window.
```
Comment (TartanLlama) [inline vcpkg/commands/edit.md:41]: Shouldn't be a comma between "`packages`, and`

File: vcpkg/commands/edit.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+If Visual Studio Code is not found, vcpkg attempts to use the default text editor configured for
+your system.
+
+## Options
+
+All vcpkg commands support a set of [common options](common-options.md).
+
+### `--all`
+
+Opens the port's related `buildtrees`, and `packages` folders.
```
Comment (TartanLlama) [inline vcpkg/commands/edit.md:68]: Shouldn't be a comma

File: vcpkg/commands/edit.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+vcpkg install fmt
+vcpkg edit fmt --all
+```
+
+## Description
+
+Opens a port for editing in a text editor window (defaults to Visual Studio Code).
+
+If multiple port names are provided, all ports are opened in the same window.
+This command does not work with ports from external registries, only ports from the built-in
```
Comment (AugP) [inline vcpkg/commands/edit.md:37]:
```suggestion
This command does not work with ports from external registries. Only ports from the built-in
```

File: vcpkg/commands/edit.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+By default, vcpkg will search for Visual Studio Code in well-known installation paths.
+Set the `EDITOR` environment variable to specify a text editor program to use.
+
+On Windows, vcpkg searches for a Visual Studio Code or Visual Studio Code Insiders installation in:
+
+* the `%ProgramFiles%` and `%ProgramFiles(x86)%` folders
+* the `%APPDATA%\Local\Programs` folder
+* the Windows Registry
+
+On Linux, vcpkg searches for the Visual Studio Code executable:
```
Comment (AugP) [inline vcpkg/commands/edit.md:52]:
```suggestion
On Linux, vcpkg searches for the Visual Studio Code executable in:
```

**Review by AugP — APPROVED** [review]

**Pull request merged**